### ISSUE TITLE
Symbol not found: _syslog$DARWIN_EXTSN with version 2.7.6

### ISSUE THREAD

**Issue by maerteijn** [psycopg/psycopg2#807]
Full stacktrace:
```
  File "/Users/martijn/.pyenv/versions/3.7.0/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/Users/martijn/.pyenv/versions/3.7.0/envs/my-virtualenv/lib/python3.7/site-packages/django/db/backends/postgresql/base.py", line 24, in <module>
    raise ImproperlyConfigured("Error loading psycopg2 module: %s" % e)
django.core.exceptions.ImproperlyConfigured: Error loading psycopg2 module: dlopen(/Users/martijn/.pyenv/versions/3.7.0/envs/my-virtualenv/lib/python3.7/site-packages/psycopg2/_psycopg.cpython-37m-darwin.so, 2): Symbol not found: _syslog$DARWIN_EXTSN
  Referenced from: /Users/martijn/.pyenv/versions/3.7.0/envs/my-virtualenv/lib/python3.7/site-packages/psycopg2/.dylibs/libcrypto.1.0.0.dylib (which was built for Mac OS X 10.13)
  Expected in: /usr/lib/libSystem.B.dylib
 in /Users/martijn/.pyenv/versions/3.7.0/envs/my-virtualenv/lib/python3.7/site-packages/psycopg2/.dylibs/libcrypto.1.0.0.dylib
```
This does not happen with the binary wheel version 2.7.5.

Installing from source (`pip install --no-binary :all: "psycopg2==2.7.6"`) does not give any problems.

OS: OSX 10.12.6
Python version: 3.7.0
pyenv: 1.2.6

Does this mean Mac OSX Sierra (10.12) is not supported with the binary wheel packages anymore? If this is on purpose, maybe this could be added to the release notes.

**Comment by dvarrazzo** [comment]
I don't know if anything changed in the way wheels are built for OSX. I didn't change anything relevant in the build chain.

FYI the Travis jobs which built the packages are [this for psycopg2](https://travis-ci.org/psycopg/psycopg2-wheels/jobs/452522369) and [this for psycopg2-binary](https://travis-ci.org/psycopg/psycopg2-wheels/jobs/452522370). They were tested but only on the same Travis machine. That's all I can do.

I don't own any OSX machine to play with. If you can suggest a fix that'd be welcome. Thank you very much.

**Comment by rjruizes** [comment]
I got the same error. I'm using OSX 10.12.6, Python 3.7

```
    from psycopg2._psycopg import (                     # noqa
ImportError: dlopen(/Users/rr214/.local/share/virtualenvs/postgresDemo-pZMzV3wg/lib/python3.7/site-packages/psycopg2/_psycopg.cpython-37m-darwin.so, 2): Symbol not found: _syslog$DARWIN_EXTSN
  Referenced from: /Users/rr214/.local/share/virtualenvs/postgresDemo-pZMzV3wg/lib/python3.7/site-packages/psycopg2/.dylibs/libcrypto.1.0.0.dylib (which was built for Mac OS X 10.13)
```

psycopg2 2.7.5 works; 2.7.6 gives the error.

(Temporary workaround `pipenv install "psycopg2-binary<=2.7.5"`)

**Comment by dvarrazzo** [comment]
So could the problem be an updated version of crypto on the Travis machines which is not compatible with OSX 10.12? Can it be solved by compiling crypto locally?

**Comment by takeda** [comment]
Not sure I understand the question but at least in my and @maerteijn's case looks like everything works fine if psycopg2 is compiled locally i.e. `pip install --no-binary :all: psycopg2` installs a working library.

**Comment by dvarrazzo** [comment]
@takeda of course: this is a problem with the binary package.

**Comment by dvarrazzo** [comment]
This is the [list of supported OSX versions on Travis](https://docs.travis-ci.com/user/reference/osx/#os-x-version). 2.7.6 [was built](https://travis-ci.org/psycopg/psycopg2-wheels/jobs/452522369#L4) using OS X 10.13 and Xcode 9.4.1. I'm building test packages using Xcode 9.2, the last available running on OS X 10.12. However, for a reference, 2.7.5 [was built](https://travis-ci.org/psycopg/psycopg2-wheels/jobs/393335126#L4) with Xcode 8.3.

**Comment by dvarrazzo** [comment]
I've made test packages [available on Test PyPI](https://test.pypi.org/project/psycopg2/2.7.6.1.dev1/) as Psycopg 2.7.6.1.dev1. Please test them using:

```
pip install -i https://test.pypi.org/simple/ psycopg2==2.7.6.1.dev1
```

and let me know if they fix the problem. Thank you.

**Comment by takeda** [comment]
That version works on my machine.

**Comment by maerteijn** [comment]
I can also confirm these test packages work!

**Comment by dvarrazzo** [comment]
Thank you all for testing. Psycopg binary packages 2.7.6.1 released.

**Comment by haeyong27** [comment]
> Thank you all for testing. Psycopg binary packages 2.7.6.1 released.

It works for me too. Thanks!

**Comment by clrengif** [comment]
works!! 2019

**Comment by bcrant** [comment]
Saves the day! So glad I found this. Still working in 2020. Installing psycopg2-binary did the trick. Thank you!

**Comment by mariuspotor** [comment]
You can also obtain a stand-alone package, not requiring a compiler or external libraries:
`pip install psycopg2-binary`

**Comment by Phil999-art** [comment]
Hello, I've tried the above solutions, neither of them work when I try to import the psycopg2 package in IDLE or ATOM. However, when I try to import it within Jupyter, it works. I install the 2.7.6.1 version, then when it didn't work, I reinstalled the 2.8.5 version. Does anyone know why it only works within Jupyter notebooks and not any other editor/IDE? I used the pip installer instead of the conda installer.
